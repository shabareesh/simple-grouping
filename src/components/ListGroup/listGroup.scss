.list-group {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 30px;
  max-width: 1000px;
  margin: 0 auto;

  &--card {
    background: #fff;
    flex-basis: calc(50% - 10px);
    margin-bottom: 20px;
    border-radius: 8px;

    &__title {
      margin: 0;
      padding: 10px;
      font-size: 20px;
      font-weight: 500;
      background: #033B00;
      color: #fff;
      border-radius: 8px 8px 0 0;
    }

    &__list {
      height: 200px;
      overflow-y: auto;
      margin: 0;
      padding: 0;
      list-style: none;

      li {
        display: block;
        padding: 15px 10px;
        border-bottom: 1px solid rgba(#033B00, 0.3);
        font-size: 14px;
      }
    }
  }
}
