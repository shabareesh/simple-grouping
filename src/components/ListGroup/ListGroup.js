import React from 'react';
import data from '../../api/data';
import { getGroupedData } from '../../utils/utils';

import './listGroup.scss';

const ListGroup = () => (
	<div className="list-group">
		{getGroupedData(data).map(({ listId, list }) => (
			<div className="list-group--card" key={listId}>
				<h4 className="list-group--card__title">List ID: {listId}</h4>
				<ul className="list-group--card__list">
					{list.map((e) => (
						<li key={e.id}>{e.name}</li>
					))}
				</ul>
			</div>
		))}
	</div>
);

export default ListGroup;
