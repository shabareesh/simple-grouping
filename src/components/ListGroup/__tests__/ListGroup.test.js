import React from 'react';
import {render} from "@testing-library/react";
import ListGroup from "../ListGroup";

describe('ListGroup', () => {
	test('should match snapshot', () => {
		const { baseElement } = render(<ListGroup />);

		expect(baseElement).toMatchSnapshot();
	});
});
