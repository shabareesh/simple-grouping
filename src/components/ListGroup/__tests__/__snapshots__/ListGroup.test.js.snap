// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup should match snapshot 1`] = `
<body>
  <div>
    <div
      class="list-group"
    >
      <div
        class="list-group--card"
      >
        <h4
          class="list-group--card__title"
        >
          List ID: 
          1
        </h4>
        <ul
          class="list-group--card__list"
        >
          <li>
            Item 0
          </li>
          <li>
            Item 4
          </li>
          <li>
            Item 28
          </li>
          <li>
            Item 29
          </li>
          <li>
            Item 39
          </li>
          <li>
            Item 41
          </li>
          <li>
            Item 50
          </li>
          <li>
            Item 54
          </li>
          <li>
            Item 70
          </li>
          <li>
            Item 85
          </li>
          <li>
            Item 88
          </li>
          <li>
            Item 90
          </li>
          <li>
            Item 101
          </li>
          <li>
            Item 110
          </li>
          <li>
            Item 136
          </li>
          <li>
            Item 147
          </li>
          <li>
            Item 153
          </li>
          <li>
            Item 157
          </li>
          <li>
            Item 181
          </li>
          <li>
            Item 189
          </li>
          <li>
            Item 193
          </li>
          <li>
            Item 196
          </li>
          <li>
            Item 200
          </li>
          <li>
            Item 201
          </li>
          <li>
            Item 211
          </li>
          <li>
            Item 224
          </li>
          <li>
            Item 238
          </li>
          <li>
            Item 262
          </li>
          <li>
            Item 269
          </li>
          <li>
            Item 276
          </li>
          <li>
            Item 280
          </li>
          <li>
            Item 342
          </li>
          <li>
            Item 376
          </li>
          <li>
            Item 395
          </li>
          <li>
            Item 397
          </li>
          <li>
            Item 413
          </li>
          <li>
            Item 414
          </li>
          <li>
            Item 415
          </li>
          <li>
            Item 447
          </li>
          <li>
            Item 451
          </li>
          <li>
            Item 456
          </li>
          <li>
            Item 458
          </li>
          <li>
            Item 473
          </li>
          <li>
            Item 479
          </li>
          <li>
            Item 485
          </li>
          <li>
            Item 496
          </li>
          <li>
            Item 520
          </li>
          <li>
            Item 524
          </li>
          <li>
            Item 536
          </li>
          <li>
            Item 545
          </li>
          <li>
            Item 548
          </li>
          <li>
            Item 568
          </li>
          <li>
            Item 587
          </li>
          <li>
            Item 600
          </li>
          <li>
            Item 617
          </li>
          <li>
            Item 620
          </li>
          <li>
            Item 627
          </li>
          <li>
            Item 635
          </li>
          <li>
            Item 642
          </li>
          <li>
            Item 662
          </li>
          <li>
            Item 678
          </li>
          <li>
            Item 684
          </li>
          <li>
            Item 688
          </li>
          <li>
            Item 727
          </li>
          <li>
            Item 735
          </li>
          <li>
            Item 787
          </li>
          <li>
            Item 790
          </li>
          <li>
            Item 791
          </li>
          <li>
            Item 799
          </li>
          <li>
            Item 805
          </li>
          <li>
            Item 815
          </li>
          <li>
            Item 825
          </li>
          <li>
            Item 827
          </li>
          <li>
            Item 838
          </li>
          <li>
            Item 861
          </li>
          <li>
            Item 891
          </li>
          <li>
            Item 912
          </li>
          <li>
            Item 920
          </li>
          <li>
            Item 923
          </li>
          <li>
            Item 929
          </li>
          <li>
            Item 930
          </li>
        </ul>
      </div>
      <div
        class="list-group--card"
      >
        <h4
          class="list-group--card__title"
        >
          List ID: 
          2
        </h4>
        <ul
          class="list-group--card__list"
        >
          <li>
            Item 3
          </li>
          <li>
            Item 6
          </li>
          <li>
            Item 21
          </li>
          <li>
            Item 43
          </li>
          <li>
            Item 65
          </li>
          <li>
            Item 67
          </li>
          <li>
            Item 105
          </li>
          <li>
            Item 122
          </li>
          <li>
            Item 139
          </li>
          <li>
            Item 143
          </li>
          <li>
            Item 148
          </li>
          <li>
            Item 154
          </li>
          <li>
            Item 160
          </li>
          <li>
            Item 164
          </li>
          <li>
            Item 168
          </li>
          <li>
            Item 182
          </li>
          <li>
            Item 188
          </li>
          <li>
            Item 197
          </li>
          <li>
            Item 221
          </li>
          <li>
            Item 255
          </li>
          <li>
            Item 259
          </li>
          <li>
            Item 261
          </li>
          <li>
            Item 265
          </li>
          <li>
            Item 296
          </li>
          <li>
            Item 329
          </li>
          <li>
            Item 339
          </li>
          <li>
            Item 341
          </li>
          <li>
            Item 344
          </li>
          <li>
            Item 364
          </li>
          <li>
            Item 388
          </li>
          <li>
            Item 390
          </li>
          <li>
            Item 394
          </li>
          <li>
            Item 420
          </li>
          <li>
            Item 430
          </li>
          <li>
            Item 434
          </li>
          <li>
            Item 440
          </li>
          <li>
            Item 452
          </li>
          <li>
            Item 497
          </li>
          <li>
            Item 512
          </li>
          <li>
            Item 546
          </li>
          <li>
            Item 573
          </li>
          <li>
            Item 578
          </li>
          <li>
            Item 592
          </li>
          <li>
            Item 619
          </li>
          <li>
            Item 639
          </li>
          <li>
            Item 656
          </li>
          <li>
            Item 669
          </li>
          <li>
            Item 690
          </li>
          <li>
            Item 725
          </li>
          <li>
            Item 739
          </li>
          <li>
            Item 763
          </li>
          <li>
            Item 777
          </li>
          <li>
            Item 778
          </li>
          <li>
            Item 798
          </li>
          <li>
            Item 831
          </li>
          <li>
            Item 843
          </li>
          <li>
            Item 875
          </li>
          <li>
            Item 886
          </li>
          <li>
            Item 887
          </li>
          <li>
            Item 890
          </li>
          <li>
            Item 900
          </li>
          <li>
            Item 901
          </li>
          <li>
            Item 906
          </li>
          <li>
            Item 910
          </li>
          <li>
            Item 928
          </li>
          <li>
            Item 955
          </li>
          <li>
            Item 970
          </li>
          <li>
            Item 973
          </li>
          <li>
            Item 978
          </li>
          <li>
            Item 980
          </li>
          <li>
            Item 996
          </li>
        </ul>
      </div>
      <div
        class="list-group--card"
      >
        <h4
          class="list-group--card__title"
        >
          List ID: 
          3
        </h4>
        <ul
          class="list-group--card__list"
        >
          <li>
            Item 5
          </li>
          <li>
            Item 12
          </li>
          <li>
            Item 19
          </li>
          <li>
            Item 26
          </li>
          <li>
            Item 27
          </li>
          <li>
            Item 32
          </li>
          <li>
            Item 47
          </li>
          <li>
            Item 68
          </li>
          <li>
            Item 71
          </li>
          <li>
            Item 74
          </li>
          <li>
            Item 75
          </li>
          <li>
            Item 99
          </li>
          <li>
            Item 102
          </li>
          <li>
            Item 132
          </li>
          <li>
            Item 137
          </li>
          <li>
            Item 151
          </li>
          <li>
            Item 176
          </li>
          <li>
            Item 192
          </li>
          <li>
            Item 202
          </li>
          <li>
            Item 208
          </li>
          <li>
            Item 218
          </li>
          <li>
            Item 226
          </li>
          <li>
            Item 235
          </li>
          <li>
            Item 241
          </li>
          <li>
            Item 256
          </li>
          <li>
            Item 275
          </li>
          <li>
            Item 310
          </li>
          <li>
            Item 324
          </li>
          <li>
            Item 326
          </li>
          <li>
            Item 331
          </li>
          <li>
            Item 356
          </li>
          <li>
            Item 368
          </li>
          <li>
            Item 374
          </li>
          <li>
            Item 387
          </li>
          <li>
            Item 406
          </li>
          <li>
            Item 416
          </li>
          <li>
            Item 417
          </li>
          <li>
            Item 433
          </li>
          <li>
            Item 459
          </li>
          <li>
            Item 464
          </li>
          <li>
            Item 466
          </li>
          <li>
            Item 471
          </li>
          <li>
            Item 476
          </li>
          <li>
            Item 511
          </li>
          <li>
            Item 533
          </li>
          <li>
            Item 543
          </li>
          <li>
            Item 556
          </li>
          <li>
            Item 559
          </li>
          <li>
            Item 565
          </li>
          <li>
            Item 581
          </li>
          <li>
            Item 601
          </li>
          <li>
            Item 602
          </li>
          <li>
            Item 604
          </li>
          <li>
            Item 654
          </li>
          <li>
            Item 661
          </li>
          <li>
            Item 666
          </li>
          <li>
            Item 680
          </li>
          <li>
            Item 686
          </li>
          <li>
            Item 693
          </li>
          <li>
            Item 694
          </li>
          <li>
            Item 699
          </li>
          <li>
            Item 709
          </li>
          <li>
            Item 710
          </li>
          <li>
            Item 712
          </li>
          <li>
            Item 734
          </li>
          <li>
            Item 759
          </li>
          <li>
            Item 816
          </li>
          <li>
            Item 833
          </li>
          <li>
            Item 858
          </li>
          <li>
            Item 894
          </li>
          <li>
            Item 905
          </li>
          <li>
            Item 907
          </li>
          <li>
            Item 914
          </li>
          <li>
            Item 916
          </li>
          <li>
            Item 918
          </li>
          <li>
            Item 934
          </li>
          <li>
            Item 941
          </li>
          <li>
            Item 942
          </li>
          <li>
            Item 956
          </li>
          <li>
            Item 960
          </li>
          <li>
            Item 971
          </li>
          <li>
            Item 975
          </li>
          <li>
            Item 984
          </li>
          <li>
            Item 985
          </li>
          <li>
            Item 994
          </li>
          <li>
            Item 998
          </li>
        </ul>
      </div>
      <div
        class="list-group--card"
      >
        <h4
          class="list-group--card__title"
        >
          List ID: 
          4
        </h4>
        <ul
          class="list-group--card__list"
        >
          <li>
            Item 1
          </li>
          <li>
            Item 8
          </li>
          <li>
            Item 24
          </li>
          <li>
            Item 25
          </li>
          <li>
            Item 60
          </li>
          <li>
            Item 69
          </li>
          <li>
            Item 73
          </li>
          <li>
            Item 76
          </li>
          <li>
            Item 83
          </li>
          <li>
            Item 89
          </li>
          <li>
            Item 95
          </li>
          <li>
            Item 114
          </li>
          <li>
            Item 119
          </li>
          <li>
            Item 161
          </li>
          <li>
            Item 186
          </li>
          <li>
            Item 206
          </li>
          <li>
            Item 207
          </li>
          <li>
            Item 236
          </li>
          <li>
            Item 273
          </li>
          <li>
            Item 284
          </li>
          <li>
            Item 318
          </li>
          <li>
            Item 320
          </li>
          <li>
            Item 335
          </li>
          <li>
            Item 345
          </li>
          <li>
            Item 350
          </li>
          <li>
            Item 384
          </li>
          <li>
            Item 393
          </li>
          <li>
            Item 399
          </li>
          <li>
            Item 426
          </li>
          <li>
            Item 427
          </li>
          <li>
            Item 450
          </li>
          <li>
            Item 463
          </li>
          <li>
            Item 469
          </li>
          <li>
            Item 475
          </li>
          <li>
            Item 482
          </li>
          <li>
            Item 484
          </li>
          <li>
            Item 493
          </li>
          <li>
            Item 523
          </li>
          <li>
            Item 525
          </li>
          <li>
            Item 534
          </li>
          <li>
            Item 551
          </li>
          <li>
            Item 555
          </li>
          <li>
            Item 564
          </li>
          <li>
            Item 575
          </li>
          <li>
            Item 576
          </li>
          <li>
            Item 594
          </li>
          <li>
            Item 597
          </li>
          <li>
            Item 633
          </li>
          <li>
            Item 641
          </li>
          <li>
            Item 645
          </li>
          <li>
            Item 650
          </li>
          <li>
            Item 653
          </li>
          <li>
            Item 663
          </li>
          <li>
            Item 681
          </li>
          <li>
            Item 697
          </li>
          <li>
            Item 701
          </li>
          <li>
            Item 702
          </li>
          <li>
            Item 714
          </li>
          <li>
            Item 715
          </li>
          <li>
            Item 726
          </li>
          <li>
            Item 737
          </li>
          <li>
            Item 740
          </li>
          <li>
            Item 749
          </li>
          <li>
            Item 752
          </li>
          <li>
            Item 754
          </li>
          <li>
            Item 766
          </li>
          <li>
            Item 772
          </li>
          <li>
            Item 784
          </li>
          <li>
            Item 794
          </li>
          <li>
            Item 808
          </li>
          <li>
            Item 817
          </li>
          <li>
            Item 823
          </li>
          <li>
            Item 824
          </li>
          <li>
            Item 847
          </li>
          <li>
            Item 879
          </li>
          <li>
            Item 908
          </li>
          <li>
            Item 927
          </li>
          <li>
            Item 937
          </li>
          <li>
            Item 950
          </li>
          <li>
            Item 958
          </li>
          <li>
            Item 962
          </li>
          <li>
            Item 969
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;
